x-pack/plugins/stack_connectors/public/connector_types/stack/opsgenie/params.tsx

```diff
@@ -85,11 +86,20 @@ const OpsgenieParamFields: React.FC<ActionParamsProps<OpsgenieActionParams>> = (
   useEffect(() => {
     if (subAction != null && currentSubAction.current !== subAction) {
       currentSubAction.current = subAction;
-      const params = subActionParams?.alias ? { alias: subActionParams.alias } : undefined;
-      editAction('subActionParams', params, index);
+
+      // check for a mismatch in the subAction and params, if the subAction does not match the params then we need to
+      // clear them by calling editAction. We can carry over the alias if it exists
+      if (
```

Comment:
Instead of always reseting the form, we can deduce if a deletion occurred by checking to see if the subAction and subActionParams are aligned (this occurs when a deletion happens where the subActions were different). If the subActionParams does not match the newly set subAction that means the user is in the Test mode and just switched the action from create alert to close alert. If the user changes the `run when` option in the rule form this code will still preserve the passed in values but the component will get rerendered later with the defaults which is fine because that's how the behavior worked before fixing the bug.